### PR TITLE
Dp lis java support

### PR DESCRIPTION
**Pull request by andrewwangva**
I'll finish the last section later.

### CODE REVIEW

File: content/4_Gold/LIS.mdx [hunk 1]
```diff
@@ -251,7 +292,7 @@ int main() {
     cin >> n;
     for (int i = 0; i < n; i++) cin >> a[i].first >> a[i].second;
     sort(a, a + n, greater<pair<int, int>>());
-
+    //finding the LIS of a reversed array = finding the LDS
```
Comment (dolphingarlic) [inline content/4_Gold/LIS.mdx:295]: This comment isn't necessary

File: content/4_Gold/LIS.mdx [hunk 2]
```diff
@@ -263,6 +304,39 @@ int main() {
 }
 ```
 
+</CPPSection>
+
+<JavaSection>
+
+```java
+class PCB{
```
Comment (dolphingarlic) [inline content/4_Gold/LIS.mdx:312]: This code doesn't compile on CSES

File: content/4_Gold/LIS.mdx [hunk 3]
```diff
@@ -163,6 +163,25 @@ int find_lis(vector<int> arr) {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+
+public static int find_lis(int[] a) {
```
Comment (dolphingarlic) [inline content/4_Gold/LIS.mdx:170]: I don't think this is the code for the O(N^2) LIS

**Review by dolphingarlic — CHANGES_REQUESTED** [review]

File: content/4_Gold/LIS.mdx (hunk 1 above)
Comment (andrewwangva) [reply content/4_Gold/LIS.mdx:297]: Why? I thought it was a little confusing when I first read through it

Comment (thecodingwizard) [reply content/4_Gold/LIS.mdx:297]: I feel like this comment is okay? Might not be obvious to everyone :P

**Review by dolphingarlic — APPROVED** [review]

**Pull request merged**